### PR TITLE
expanded instructions in README

### PR DESCRIPTION
**Pull request by stevengj**
This expands the README instructions to discuss plugin installation, compilation with `cmake`, the required `H5P` calls, etc.

Note, however, that this PR assumes that #3 has been fixed (so that the `cmake` script builds the filter plugin too).

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -11,18 +11,43 @@ Blosc filter for HDF5
...
-================================
+Instead of just linking this Blosc filter into your HDF5 application, it is possible to install
+it as a system-wide HDF5 plugin (with HDF5 1.8.11 or later).  This is useful because it allows
+*every* HDF5-using program on your system to transparently read Blosc-compressed HDF5 files.
 
-In order to register Blosc into your HDF5 application, you only need
+As described in the `HDF5 plugin documentation <https://www.hdfgroup.org/HDF5/doc/Advanced/DynamicallyLoadedFilters/HDF5DynamicallyLoadedFilters.pdf>`_, you just need to compile the Blosc plugin into a shared library and
+copy it to the plugin directory (which defaults to ``/usr/local/hdf5/lib/plugin`` on non-Windows systems).
+
+Following the ``cmake`` instructions below produces a ``blosc_plugin.so`` shared library
```
Comment (aldanor) [inline README.rst:32]: It doesn't look like `cmake` build generates the plugin shared library, only the filter.

Comment (stevengj) [inline README.rst:32]: See the note above about #3

Comment (aldanor) [inline README.rst:32]: Gotcha. Makes sense then.

**Comment by FrancescAlted** [comment]
What's the status of this?  We want to merge this or should we wait until the situation of the filter as plugin is clearer?

**Comment by stevengj** [comment]
I was hoping to have time to make `cmake` build the plugin, but I haven't gotten around to it... can't really merge this until someone does this (which really should be done... the plugin is pretty useful).

**Comment by estan** [comment]
With #3 now closed (plugin is built), could this be merged? Just like @toddrme2178 in #9 , I would love a release so this can be packaged.

**Comment by FrancescAlted** [comment]
Good point @estan .

**Pull request merged**